src/utils/authenticateUser.ts

```diff
@@ -0,0 +1,20 @@
+import axios from 'axios';
+
+const API_URL_PROD = 'https://api.example.com';
+
+export const authenticateUser = async (
+  username: string,
+  password: string
+): Promise<string> => {
+  const response = await axios.post(`${API_URL_PROD}/token`, {
+    username,
+    password
+  });
+
+  //check if the response is valid
+  if (response.status !== 200) {
+    throw new Error('Invalid response from server!');
```

Comment:
I'd suggest wrapping the axios call in a try catch block here and always returning a valid response from the API functions. That will make usage of the API functions simpler in the components.